automate/serializers.py

```diff
@@ -0,0 +1,49 @@
+from rest_framework import serializers
+from django.core.exceptions import ValidationError
+from .models import Project
+from repo.utils import add_hook_to_repo, gen_hook_url
+
+
+class ProjectSerializer(serializers.ModelSerializer):
+    """Repository Serializer."""
+    token = serializers.CharField(write_only=True)
+    user = serializers.SerializerMethodField()
+
+    class Meta:
+        """Meta class for Repository Serializer."""
+
+        model = Project
+        exclude = ["created_at", "updated_at"]
+        lookup_field = "slug"
+        extra_kwargs = {"owner": {"read_only": True}}
+
+    def create(self, validated_data):
+        repo = super.create(validated_data)
+        if not repo:
+            raise ValidationError({'error': 'this repo bundle was not initialized'})
+        try:
+            repo_name = validated_data['primary_repo']
+            repo_name = str(repo_name).split('/')
+            repo_name = repo_name[-1]
+            repo_name = repo_name.split('.')[0]
```

Comment:
Really confusing, Maybe explaining what is happening here would help as comments.
Also `repo_name`, `repo_name`,  what happen? Give proper names